client/components/data/query-atat-eligibility/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { flowRight as compose } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { requestEligibility } from 'state/automated-transfer/actions';
+
+export class QueryAutomatedTransferEligibility extends Component {
+	static propTypes = {
+		request: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+	};
+
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( { request, siteId } ) {
+		request( siteId );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+export const mapDispatchToProps = dispatch => ( {
+	request: compose( dispatch, requestEligibility ),
+} );
+
+export default connect( null, mapDispatchToProps )( QueryAutomatedTransferEligibility );
```

Comment:
Just fyi for the object value: ```If an object is passed, each function inside it will be assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props.``` [docs](https://github.com/reactjs/react-redux/blob/master/docs/api.md#arguments) 

I personally prefer the object shorthand, if we don't need custom behavior.